handlers/signature_request_test.go

```diff
@@ -0,0 +1,97 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package handlers
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+	"github.com/ava-labs/avalanchego/vms/platformvm/teleporter"
+	"github.com/ava-labs/subnet-evm/plugin/evm/message"
+	"github.com/ava-labs/subnet-evm/plugin/evm/warp"
+	"github.com/ava-labs/subnet-evm/sync/handlers/stats"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSignatureHandler(t *testing.T) {
```

Comment:
Could we add a unit test to `plugin/evm/vm_test.go` as well that verifies sending a request with this code is dropped correctly and does not result in a panic as a regression for the previous version that left the warp backend unpopulated? We should add a TODO to modify the test when we actually activate warp so that the expected behavior changes from dropping without a panic or error to responding correctly.

This should be a good starting point for what that unit test should look like: https://github.com/ava-labs/subnet-evm/blob/master/plugin/evm/vm_test.go#L2995.

The test should:

- Set up the VM
- Create a signature request for a non-existent signature
- Call `vm.AppRequest(...)` with the message and ensure that it does not return an error or panic
- Include a TODO to update the test when the signature request handler should be activated

Reply:
Bumping this

Reply:
just pushed the `vm_test` but figuring out fixing tree formats